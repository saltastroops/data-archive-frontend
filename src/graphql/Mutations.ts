import gql from "graphql-tag";

/**
 * Create a new user.
 *
 * The following arguments must be supplied.
 *
 * affiliation:
 *     The affiliation of the user, such as a university or an institute.
 * email:
 *     Email address. This will be stored in lower case.
 * familyName:
 *     The family name (surname).
 * givenName:
 *     The given name (first name).
 * password:
 *     The password, which must have at least 7 characters.
 * username:
 *     The username, which must not contain upper case letters.
 */
export const SIGNUP_MUTATION = gql`
  mutation SIGNUP_MUTATION(
    $familyName: String!
    $givenName: String!
    $username: String!
    $email: String!
    $affiliation: String!
    $password: String!
  ) {
    signup(
      familyName: $familyName
      givenName: $givenName
      username: $username
      email: $email
      affiliation: $affiliation
      password: $password
    ) {
      id
    }
  }
`;

/**
 * Log the user in.
 *
 * The following arguments must be supplied
 *
 * password:
 *     The password.
 * username:
 *     The uusername.
 */
export const LOGIN_MUTATION = gql`
  mutation LOGIN_MUTATION($username: String!, $password: String!) {
    login(password: $password, username: $username) @client
  }
`;

/**
 * Log the user out.
 */
export const LOGOUT_MUTATION = gql`
  mutation LOGOUT_MUTATION {
    logout @client
  }
`;

/**
 * Update user.
 *
 * The following arguments may be supplied.
 *
 * affiliation:
 *     The affiliation of the user, such as a university or an institute.
 * email:
 *     Email address. This will be stored in lower case.
 * familyName:
 *     The family name (surname).
 * givenName:
 *     The given name (first name).
 * id:
 *     The user ID different from the currently logged in user.
 * newPassword:
 *     The new user password, which must have at lease 7 characters.
 * password:
 *     The password, which must have at least 7 characters and is required.
 * username:
 *     The username, which must not contain upper case letters.
 */
export const UPDATE_USER_MUTATION = gql`
  mutation UPDATE_USER_MUTATION(
    $id: ID
    $familyName: String
    $givenName: String
    $username: String
    $email: String
    $affiliation: String
    $password: String!
    $newPassword: String
  ) {
    updateUser(
      id: $id
      familyName: $familyName
      givenName: $givenName
      username: $username
      email: $email
      affiliation: $affiliation
      password: $password
      newPassword: $newPassword
    ) {
      id
    }
  }
`;
