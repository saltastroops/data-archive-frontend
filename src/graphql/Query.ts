import gql from "graphql-tag";

// TODO UPDATE
/**
 * Retrieving the user information.
 */
const USER_QUERY = gql`
  query USER_QUERY {
    user {
      familyName
      givenName
      roles
    }
  }
`;

export { USER_QUERY };
