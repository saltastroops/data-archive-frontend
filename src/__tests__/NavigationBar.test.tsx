import { mount } from "enzyme";
import toJSON from "enzyme-to-json";
import * as React from "react";
import { MockedProvider } from "react-apollo/test-utils";
import { MemoryRouter } from "react-router";
import wait from "waait";
import NavigationBar from "../components/NavigationBar";
import { LOGOUT_MUTATION } from "../graphql/Mutations";
import { USER_QUERY } from "../graphql/Query";

const mockUser = (familyName: string, givenName: string, isAdmin: boolean) => ({
  familyName,
  givenName,
  isAdmin
});

describe("NavigationBar", () => {
  it("shows the correct content when the user is not logged in", () => {
    const user = null;

    const wrapper = mount(
      <MockedProvider>
        <MemoryRouter>
          <NavigationBar user={user} />
        </MemoryRouter>
      </MockedProvider>
    );

    // Login button is shown
    expect(wrapper.exists('Link[to="/login"]')).toBeTruthy();

    // Registration button is shown
    expect(wrapper.exists('Link[to="/register"]')).toBeTruthy();

    // Data requests link is not shown
    expect(wrapper.exists('Link[to="/data-requests"]')).toBeFalsy();

    // Account details link is not shown
    expect(wrapper.exists('Link[to="/account"]')).toBeFalsy();

    // Logout link is not shown
    expect(wrapper.exists('a[data-test="logout"]')).toBeFalsy();

    // Admin page link is not shown
    expect(wrapper.exists('Link[to="/admin"]')).toBeFalsy();

    // Overall content check
    const navigationBar = wrapper.find(".navbar");
    expect(toJSON(navigationBar)).toMatchSnapshot();
  });

  it("shows the correct content when the user is a non-administrator", () => {
    const user = mockUser("surname", "name", false);

    const wrapper = mount(
      <MockedProvider>
        <MemoryRouter>
          <NavigationBar user={user} />
        </MemoryRouter>
      </MockedProvider>
    );

    // Login button is not shown
    expect(wrapper.exists('Link[to="/login"]')).toBeFalsy();

    // Registration button is not shown
    expect(wrapper.exists('Link[to="/register"]')).toBeFalsy();

    // Data requests link is shown
    expect(wrapper.exists('Link[to="/data-requests"]')).toBeTruthy();

    // Account details link is shown
    expect(wrapper.exists('Link[to="/account"]')).toBeTruthy();

    // Logout link is shown
    expect(wrapper.exists('a[data-test="logout"]')).toBeTruthy();

    // Admin page link is not shown
    expect(wrapper.exists('Link[to="/admin"]')).toBeFalsy();

    // Overall content check
    const navigationBar = wrapper.find(".navbar");
    expect(toJSON(navigationBar)).toMatchSnapshot();
  });

  it("shows the correct content when the user is an administrator", () => {
    const user = mockUser("surname", "name", true);

    const wrapper = mount(
      <MockedProvider>
        <MemoryRouter>
          <NavigationBar user={user} />
        </MemoryRouter>
      </MockedProvider>
    );

    // Login button is not shown
    expect(wrapper.exists('Link[to="/login"]')).toBeFalsy();

    // Registration button is not shown
    expect(wrapper.exists('Link[to="/register"]')).toBeFalsy();

    // Data requests link is shown
    expect(wrapper.exists('Link[to="/data-requests"]')).toBeTruthy();

    // Account details link is shown
    expect(wrapper.exists('Link[to="/account"]')).toBeTruthy();

    // Logout link is shown
    expect(wrapper.exists('a[data-test="logout"]')).toBeTruthy();

    // Admin page link is shown
    expect(wrapper.exists('Link[to="/admin"]')).toBeTruthy();

    // Overall content check
    const navigationBar = wrapper.find(".navbar");
    expect(toJSON(navigationBar)).toMatchSnapshot();
  });

  it("calls the logout function when the logout link is clicked", async () => {
    const user = mockUser("surname", "name", false);

    // logout mock mutation
    let logoutCalled = false;
    const mocks = [
      {
        request: {
          query: LOGOUT_MUTATION
        },
        result: () => {
          logoutCalled = true;
          return {
            data: {
              success: true
            }
          };
        }
      },

      {
        request: {
          query: USER_QUERY
        },
        result: {
          data: {
            user: null
          }
        }
      }
    ];

    const wrapper = mount(
      <MockedProvider mocks={mocks}>
        <MemoryRouter>
          <NavigationBar user={user} />
        </MemoryRouter>
      </MockedProvider>
    );

    // The logout mutation has not been called (yet).
    expect(logoutCalled).toBe(false);

    // Click the logout link
    const logoutLink = wrapper.find('a[data-test="logout"]');
    logoutLink.simulate("click");

    // The logout mutation has been called
    await wait(0);
    wrapper.update();
    expect(logoutCalled).toBe(true);
  });

  it("toggles the visibility of the menu when the burger button is clicked", async () => {
    const user = mockUser("surname", "name", false);

    const wrapper = mount(
      <MockedProvider>
        <MemoryRouter>
          <NavigationBar user={user} />
        </MemoryRouter>
      </MockedProvider>
    );

    // The burger button and the menu are not active
    let burger = wrapper.find(".burger");
    let menu = wrapper.find(".navbar-menu");
    expect(burger.hasClass("is-active")).toBeFalsy();
    expect(menu.hasClass("is-active")).toBeFalsy();

    // Click the burger
    burger.simulate("click");

    // The burger button and the menu are active now
    burger = wrapper.find(".burger");
    menu = wrapper.find(".navbar-menu");
    expect(burger.hasClass("is-active")).toBeTruthy();
    expect(menu.hasClass("is-active")).toBeTruthy();

    // Click the burger again
    burger.simulate("click");

    // The burger button and the menu are inactive again
    burger = wrapper.find(".burger");
    menu = wrapper.find(".navbar-menu");
    expect(burger.hasClass("is-active")).toBeFalsy();
    expect(menu.hasClass("is-active")).toBeFalsy();
  });
});
