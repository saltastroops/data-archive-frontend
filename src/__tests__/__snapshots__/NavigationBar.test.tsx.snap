// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavigationBar shows the correct content when the user is a non-administrator 1`] = `
<nav
  className="navbar is-full-width  "
>
  <div
    className="navbar-brand"
  >
    <a
      aria-expanded="false"
      aria-label="menu"
      className="navbar-burger burger "
      onClick={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </a>
  </div>
  <div
    className="navbar-menu has-text-weight-light  "
  >
    <div
      className="navbar-start "
    >
      <NavLink
        activeClassName="has-text-weight-bold has-text-link"
        className="navbar-item item"
        exact={true}
        to="/"
      >
        <Route
          exact={true}
          path="\\\\/"
        >
          <Link
            aria-current="page"
            className="navbar-item item has-text-weight-bold has-text-link"
            style={Object {}}
            to="/"
          >
            <a
              aria-current="page"
              className="navbar-item item has-text-weight-bold has-text-link"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Search
            </a>
          </Link>
        </Route>
      </NavLink>
      <NavLink
        activeClassName="has-text-weight-bold has-text-link"
        className="navbar-item item"
        to="/data-requests"
      >
        <Route
          path="\\\\/data-requests"
        >
          <Link
            aria-current={null}
            className="navbar-item item"
            to="/data-requests"
          >
            <a
              aria-current={null}
              className="navbar-item item"
              href="/data-requests"
              onClick={[Function]}
            >
              Data Requests
            </a>
          </Link>
        </Route>
      </NavLink>
    </div>
    <div
      className="navbar-end subtitle is-4"
    >
      <div
        className="navbar-item has-dropdown is-hoverable"
      >
        <a
          className="navbar-link"
        >
          Hello 
          name
        </a>
        <div
          className="navbar-dropdown"
        >
          <NavLink
            activeClassName="has-text-weight-bold has-text-link"
            className="navbar-item"
            to="/account"
          >
            <Route
              path="\\\\/account"
            >
              <Link
                aria-current={null}
                className="navbar-item"
                to="/account"
              >
                <a
                  aria-current={null}
                  className="navbar-item"
                  href="/account"
                  onClick={[Function]}
                >
                  Account
                </a>
              </Link>
            </Route>
          </NavLink>
          <NavLink
            activeClassName="has-text-weight-bold has-text-link"
            className="navbar-item"
            to="/user-update"
          >
            <Route
              path="\\\\/user-update"
            >
              <Link
                aria-current={null}
                className="navbar-item"
                to="/user-update"
              >
                <a
                  aria-current={null}
                  className="navbar-item"
                  href="/user-update"
                  onClick={[Function]}
                >
                  Account Edit
                </a>
              </Link>
            </Route>
          </NavLink>
          <NavLink
            className="navbar-item"
            data-test="logout"
            onClick={[Function]}
            to="/"
          >
            <Route
              path="\\\\/"
            >
              <Link
                aria-current="page"
                className="navbar-item active"
                data-test="logout"
                onClick={[Function]}
                style={Object {}}
                to="/"
              >
                <a
                  aria-current="page"
                  className="navbar-item active"
                  data-test="logout"
                  href="/"
                  onClick={[Function]}
                  style={Object {}}
                >
                  Logout
                </a>
              </Link>
            </Route>
          </NavLink>
        </div>
      </div>
      <div
        className="navbar-item is-hidden-descktop"
      >
        <NavLink
          className="button is-link is-outlined"
          to="/cart"
        >
          <Route
            path="\\\\/cart"
          >
            <Link
              aria-current={null}
              className="button is-link is-outlined"
              to="/cart"
            >
              <a
                aria-current={null}
                className="button is-link is-outlined"
                href="/cart"
                onClick={[Function]}
              >
                <span>
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          576,
                          512,
                          Array [],
                          "f07a",
                          "M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z",
                        ],
                        "iconName": "shopping-cart",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-shopping-cart fa-w-18 "
                      data-icon="shopping-cart"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                   CART
                </span>
              </a>
            </Link>
          </Route>
        </NavLink>
      </div>
    </div>
  </div>
</nav>
`;

exports[`NavigationBar shows the correct content when the user is an administrator 1`] = `
<nav
  className="navbar is-full-width  "
>
  <div
    className="navbar-brand"
  >
    <a
      aria-expanded="false"
      aria-label="menu"
      className="navbar-burger burger "
      onClick={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </a>
  </div>
  <div
    className="navbar-menu has-text-weight-light  "
  >
    <div
      className="navbar-start "
    >
      <NavLink
        activeClassName="has-text-weight-bold has-text-link"
        className="navbar-item item"
        exact={true}
        to="/"
      >
        <Route
          exact={true}
          path="\\\\/"
        >
          <Link
            aria-current="page"
            className="navbar-item item has-text-weight-bold has-text-link"
            style={Object {}}
            to="/"
          >
            <a
              aria-current="page"
              className="navbar-item item has-text-weight-bold has-text-link"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Search
            </a>
          </Link>
        </Route>
      </NavLink>
      <NavLink
        activeClassName="has-text-weight-bold has-text-link"
        className="navbar-item item"
        to="/data-requests"
      >
        <Route
          path="\\\\/data-requests"
        >
          <Link
            aria-current={null}
            className="navbar-item item"
            to="/data-requests"
          >
            <a
              aria-current={null}
              className="navbar-item item"
              href="/data-requests"
              onClick={[Function]}
            >
              Data Requests
            </a>
          </Link>
        </Route>
      </NavLink>
      <NavLink
        activeClassName="has-text-weight-bold has-text-link"
        className="navbar-item"
        to="/admin"
      >
        <Route
          path="\\\\/admin"
        >
          <Link
            aria-current={null}
            className="navbar-item"
            to="/admin"
          >
            <a
              aria-current={null}
              className="navbar-item"
              href="/admin"
              onClick={[Function]}
            >
              Admin
            </a>
          </Link>
        </Route>
      </NavLink>
    </div>
    <div
      className="navbar-end subtitle is-4"
    >
      <div
        className="navbar-item has-dropdown is-hoverable"
      >
        <a
          className="navbar-link"
        >
          Hello 
          name
        </a>
        <div
          className="navbar-dropdown"
        >
          <NavLink
            activeClassName="has-text-weight-bold has-text-link"
            className="navbar-item"
            to="/account"
          >
            <Route
              path="\\\\/account"
            >
              <Link
                aria-current={null}
                className="navbar-item"
                to="/account"
              >
                <a
                  aria-current={null}
                  className="navbar-item"
                  href="/account"
                  onClick={[Function]}
                >
                  Account
                </a>
              </Link>
            </Route>
          </NavLink>
          <NavLink
            activeClassName="has-text-weight-bold has-text-link"
            className="navbar-item"
            to="/user-update"
          >
            <Route
              path="\\\\/user-update"
            >
              <Link
                aria-current={null}
                className="navbar-item"
                to="/user-update"
              >
                <a
                  aria-current={null}
                  className="navbar-item"
                  href="/user-update"
                  onClick={[Function]}
                >
                  Account Edit
                </a>
              </Link>
            </Route>
          </NavLink>
          <NavLink
            className="navbar-item"
            data-test="logout"
            onClick={[Function]}
            to="/"
          >
            <Route
              path="\\\\/"
            >
              <Link
                aria-current="page"
                className="navbar-item active"
                data-test="logout"
                onClick={[Function]}
                style={Object {}}
                to="/"
              >
                <a
                  aria-current="page"
                  className="navbar-item active"
                  data-test="logout"
                  href="/"
                  onClick={[Function]}
                  style={Object {}}
                >
                  Logout
                </a>
              </Link>
            </Route>
          </NavLink>
        </div>
      </div>
      <div
        className="navbar-item is-hidden-descktop"
      >
        <NavLink
          className="button is-link is-outlined"
          to="/cart"
        >
          <Route
            path="\\\\/cart"
          >
            <Link
              aria-current={null}
              className="button is-link is-outlined"
              to="/cart"
            >
              <a
                aria-current={null}
                className="button is-link is-outlined"
                href="/cart"
                onClick={[Function]}
              >
                <span>
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          576,
                          512,
                          Array [],
                          "f07a",
                          "M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z",
                        ],
                        "iconName": "shopping-cart",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-shopping-cart fa-w-18 "
                      data-icon="shopping-cart"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                   CART
                </span>
              </a>
            </Link>
          </Route>
        </NavLink>
      </div>
    </div>
  </div>
</nav>
`;

exports[`NavigationBar shows the correct content when the user is not logged in 1`] = `
<nav
  className="navbar is-full-width  "
>
  <div
    className="navbar-brand"
  >
    <a
      aria-expanded="false"
      aria-label="menu"
      className="navbar-burger burger "
      onClick={[Function]}
      role="button"
    >
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
      <span
        aria-hidden="true"
      />
    </a>
  </div>
  <div
    className="navbar-menu has-text-weight-light  "
  >
    <div
      className="navbar-start "
    >
      <NavLink
        activeClassName="has-text-weight-bold has-text-link"
        className="navbar-item item"
        exact={true}
        to="/"
      >
        <Route
          exact={true}
          path="\\\\/"
        >
          <Link
            aria-current="page"
            className="navbar-item item has-text-weight-bold has-text-link"
            style={Object {}}
            to="/"
          >
            <a
              aria-current="page"
              className="navbar-item item has-text-weight-bold has-text-link"
              href="/"
              onClick={[Function]}
              style={Object {}}
            >
              Search
            </a>
          </Link>
        </Route>
      </NavLink>
    </div>
    <div
      className="navbar-end subtitle is-4"
    >
      <div
        className="navbar-item"
      >
        <NavLink
          className=" button is-primary is-outlined"
          to="/login"
        >
          <Route
            path="\\\\/login"
          >
            <Link
              aria-current={null}
              className=" button is-primary is-outlined"
              to="/login"
            >
              <a
                aria-current={null}
                className=" button is-primary is-outlined"
                href="/login"
                onClick={[Function]}
              >
                <span>
                  Login 
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          512,
                          512,
                          Array [],
                          "f2f6",
                          "M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z",
                        ],
                        "iconName": "sign-in-alt",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-sign-in-alt fa-w-16 "
                      data-icon="sign-in-alt"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 512 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M416 448h-84c-6.6 0-12-5.4-12-12v-40c0-6.6 5.4-12 12-12h84c17.7 0 32-14.3 32-32V160c0-17.7-14.3-32-32-32h-84c-6.6 0-12-5.4-12-12V76c0-6.6 5.4-12 12-12h84c53 0 96 43 96 96v192c0 53-43 96-96 96zm-47-201L201 79c-15-15-41-4.5-41 17v96H24c-13.3 0-24 10.7-24 24v96c0 13.3 10.7 24 24 24h136v96c0 21.5 26 32 41 17l168-168c9.3-9.4 9.3-24.6 0-34z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                </span>
              </a>
            </Link>
          </Route>
        </NavLink>
      </div>
      <div
        className="navbar-item"
      >
        <NavLink
          className=" button is-info is-outlined"
          to="/register"
        >
          <Route
            path="\\\\/register"
          >
            <Link
              aria-current={null}
              className=" button is-info is-outlined"
              to="/register"
            >
              <a
                aria-current={null}
                className=" button is-info is-outlined"
                href="/register"
                onClick={[Function]}
              >
                <span>
                  Register 
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          640,
                          512,
                          Array [],
                          "f234",
                          "M624 208h-64v-64c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v64h-64c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h64v64c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-64h64c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm-400 48c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z",
                        ],
                        "iconName": "user-plus",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-user-plus fa-w-20 "
                      data-icon="user-plus"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 640 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M624 208h-64v-64c0-8.8-7.2-16-16-16h-32c-8.8 0-16 7.2-16 16v64h-64c-8.8 0-16 7.2-16 16v32c0 8.8 7.2 16 16 16h64v64c0 8.8 7.2 16 16 16h32c8.8 0 16-7.2 16-16v-64h64c8.8 0 16-7.2 16-16v-32c0-8.8-7.2-16-16-16zm-400 48c70.7 0 128-57.3 128-128S294.7 0 224 0 96 57.3 96 128s57.3 128 128 128zm89.6 32h-16.7c-22.2 10.2-46.9 16-72.9 16s-50.6-5.8-72.9-16h-16.7C60.2 288 0 348.2 0 422.4V464c0 26.5 21.5 48 48 48h352c26.5 0 48-21.5 48-48v-41.6c0-74.2-60.2-134.4-134.4-134.4z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                </span>
              </a>
            </Link>
          </Route>
        </NavLink>
      </div>
      <div
        className="navbar-item is-hidden-descktop"
      >
        <NavLink
          className="button is-link is-outlined"
          to="/cart"
        >
          <Route
            path="\\\\/cart"
          >
            <Link
              aria-current={null}
              className="button is-link is-outlined"
              to="/cart"
            >
              <a
                aria-current={null}
                className="button is-link is-outlined"
                href="/cart"
                onClick={[Function]}
              >
                <span>
                  <FontAwesomeIcon
                    border={false}
                    className=""
                    fixedWidth={false}
                    flip={null}
                    icon={
                      Object {
                        "icon": Array [
                          576,
                          512,
                          Array [],
                          "f07a",
                          "M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z",
                        ],
                        "iconName": "shopping-cart",
                        "prefix": "fas",
                      }
                    }
                    inverse={false}
                    listItem={false}
                    mask={null}
                    pull={null}
                    pulse={false}
                    rotation={null}
                    size={null}
                    spin={false}
                    symbol={false}
                    title=""
                    transform={null}
                  >
                    <svg
                      aria-hidden="true"
                      className="svg-inline--fa fa-shopping-cart fa-w-18 "
                      data-icon="shopping-cart"
                      data-prefix="fas"
                      focusable="false"
                      role="img"
                      style={Object {}}
                      viewBox="0 0 576 512"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        d="M528.12 301.319l47.273-208C578.806 78.301 567.391 64 551.99 64H159.208l-9.166-44.81C147.758 8.021 137.93 0 126.529 0H24C10.745 0 0 10.745 0 24v16c0 13.255 10.745 24 24 24h69.883l70.248 343.435C147.325 417.1 136 435.222 136 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-15.674-6.447-29.835-16.824-40h209.647C430.447 426.165 424 440.326 424 456c0 30.928 25.072 56 56 56s56-25.072 56-56c0-22.172-12.888-41.332-31.579-50.405l5.517-24.276c3.413-15.018-8.002-29.319-23.403-29.319H218.117l-6.545-32h293.145c11.206 0 20.92-7.754 23.403-18.681z"
                        fill="currentColor"
                        style={Object {}}
                      />
                    </svg>
                  </FontAwesomeIcon>
                   CART
                </span>
              </a>
            </Link>
          </Route>
        </NavLink>
      </div>
    </div>
  </div>
</nav>
`;
